zebrad/src/commands/seed.rs

```diff
@@ -0,0 +1,185 @@
+use zebra_network::{AddressBook, BoxedStdError, Request, Response};
+
+use crate::{config::ZebradConfig, prelude::*};
+
+/// Whether our `SeedService` is poll_ready or not.
+#[derive(Debug)]
+enum SeederState {
+    /// Waiting for the address book to be shared with us via the oneshot channel.
+    AwaitingAddressBook(oneshot::Receiver<Arc<Mutex<AddressBook>>>),
+    /// Address book received, ready to service requests.
+    Ready(Arc<Mutex<AddressBook>>),
+}
+
+#[derive(Debug)]
+struct SeedService {
+    state: SeederState,
+}
+
+impl Service<Request> for SeedService {
+    type Response = Response;
+    type Error = BoxedStdError;
+    type Future =
+        Pin<Box<dyn Future<Output = Result<Self::Response, Self::Error>> + Send + 'static>>;
+
+    fn poll_ready(&mut self, _cx: &mut Context<'_>) -> Poll<Result<(), Self::Error>> {
+        debug!("SeedService.state: {:?}", self.state);
+
+        match self.state {
+            SeederState::Ready(_) => return Poll::Ready(Ok(())),
+            SeederState::AwaitingAddressBook(ref mut rx) => match rx.try_recv() {
+                Err(e) => {
+                    error!("SeedService oneshot sender dropped: {:?}", e);
+                    return Poll::Ready(Err(e.into()));
+                }
+                Ok(None) => {
+                    trace!("SeedService hasn't received a message via the oneshot yet.");
+                    return Poll::Pending;
+                }
+                Ok(Some(address_book)) => {
+                    info!(
+                        "SeedService received address_book via oneshot {:?}",
+                        address_book
+                    );
+                    self.state = SeederState::Ready(address_book);
+                    return Poll::Ready(Ok(()));
+                }
+            },
+        }
+    }
+
+    fn call(&mut self, req: Request) -> Self::Future {
+        info!("SeedService handling a request: {:?}", req);
+
+        let response = match (req, &self.state) {
+            (Request::GetPeers, SeederState::Ready(address_book)) => {
+                debug!(
+                    "address_book.len(): {:?}",
+                    address_book.lock().unwrap().len()
+                );
+                info!("SeedService responding to GetPeers");
+                Ok::<Response, Self::Error>(Response::Peers(
+                    address_book.lock().unwrap().peers().collect(),
+                ))
+            }
+            _ => Ok::<Response, Self::Error>(Response::Ok),
+        };
+
+        info!("SeedService response: {:?}", response);
+        return Box::pin(futures::future::ready(response));
+    }
+}
+
+/// `seed` subcommand
+///
+/// A DNS seeder command to spider and collect as many valid peer
+/// addresses as we can.
+#[derive(Command, Debug, Default, Options)]
+pub struct SeedCmd {
+    /// Filter strings
+    #[options(free)]
+    filters: Vec<String>,
+}
+
+impl config::Override<ZebradConfig> for SeedCmd {
```

Comment:
Is there a reason that we want to be able to custom-override the tracing filters but only for the `SeedCmd`? We already have a config option and an HTTP endpoint. If not, it seems like we could cut out this whole `impl` and remove the `filters` field from `SeedCmd`.